app/models/account.rb

```diff
@@ -580,6 +580,13 @@ def generate_site_access_code
     self.site_access_code ||= SecureRandom.hex(5) if provider?
   end
 
+  def set_default_master
+    if self.provider? && self.master == nil
```

Comment:
```suggestion
    if self.provider? && !self.master?
```

I think this is safer than nil check

Reply:
I might be missing something but, wouldn't `!self.master?` evaluate to `true` if `master` is `nil` or `false`? In that case it would only be redundant in comparison to `self.master.nil?` as the body of the `if` statement sets the value to `false`.
I'm new to Ruby so there's a good chance I'll stand corrected 🙂

Reply:
As `master` is a table column, it gets handy `StringInquirer` from ActiveRecord.
So anything that is _falsey_ value (`nil`,`false`, `''`) would return `false` and any _truthy_ value will return `true`

Here we want to check that the account is a provider but not master.

No worries about being new to Ruby ;) it is a perk from ActiveSupport https://api.rubyonrails.org/classes/ActiveSupport/StringInquirer.html

Reply:
As `master` is a table column, it gets handy `StringInquirer` from ActiveRecord.
So anything that is _falsey_ value (`nil`,`false`, `''`) would return `false` and any _truthy_ value will return `true`

Here we want to check that the account is a provider but not master.

No worries about being new to Ruby ;) it is a perk from ActiveSupport https://api.rubyonrails.org/classes/ActiveSupport/StringInquirer.html

Reply:
> Here we want to check that the account is a provider **but not master**.

Actually what I want is to check that the account is a provider **but the master field is not set** (`NULL` in the database, which is what causes the triggers not to work as expected), and then set it to `false` by default.